src/main/java/org/jabref/gui/entryeditor/FieldsEditorTab.java

```diff
@@ -0,0 +1,326 @@
+                           boolean addKeyField, boolean compressed, BibEntry entry) {
+        this.entry = Objects.requireNonNull(entry);
+        this.fields = new ArrayList<>(Objects.requireNonNull(fields));
+
+        // Add the edit field for Bibtex-key.
+        if (addKeyField) {
+            this.fields.add(BibEntry.KEY_FIELD);
+        }
+
+        this.parent = parent;
+        this.frame = frame;
+        this.basePanel = basePanel;
+
+        panel = setupPanel(frame, basePanel, compressed);
+
+        // The following line makes sure focus cycles inside tab instead of being lost to other parts of the frame:
+        //panel.setFocusCycleRoot(true);
+    }
+
+    private static void addColumn(GridPane gridPane, int columnIndex, List<Label> nodes) {
+        gridPane.addColumn(columnIndex, nodes.toArray(new Node[nodes.size()]));
+    }
+
+    private static void addColumn(GridPane gridPane, int columnIndex, Stream<Parent> nodes) {
+        gridPane.addColumn(columnIndex, nodes.toArray(Node[]::new));
+    }
+
+    private String convertToHex(java.awt.Color color) {
+        return String.format("#%02x%02x%02x", color.getRed(), color.getGreen(), color.getBlue());
+    }
+
+    private Region setupPanel(JabRefFrame frame, BasePanel bPanel, boolean compressed) {
+
+        //setupKeyBindings(panel.getInputMap(JComponent.WHEN_FOCUSED), panel.getActionMap());
+
+        editors.clear();
+        List<Label> labels = new ArrayList<>();
+
+        for (String fieldName : fields) {
+
+            // TODO: Reenable/migrate this
+            // Store the editor for later reference:
+            /*
+            FieldEditor fieldEditor;
+            int defaultHeight;
+            int wHeight = (int) (50.0 * InternalBibtexFields.getFieldWeight(field));
+            if (InternalBibtexFields.getFieldProperties(field).contains(FieldProperty.SINGLE_ENTRY_LINK)) {
+                fieldEditor = new EntryLinkListEditor(frame, bPanel.getBibDatabaseContext(), field, null, parent,
+                        true);
+                defaultHeight = 0;
+            } else if (InternalBibtexFields.getFieldProperties(field).contains(FieldProperty.MULTIPLE_ENTRY_LINK)) {
+                fieldEditor = new EntryLinkListEditor(frame, bPanel.getBibDatabaseContext(), field, null, parent,
+                        false);
+                defaultHeight = 0;
+            } else {
+                fieldEditor = new TextArea(field, null, getPrompt(field));
+                //parent.addSearchListener((TextArea) fieldEditor);
+                defaultHeight = fieldEditor.getPane().getPreferredSize().height;
+            }
+            
+            Optional<JComponent> extra = parent.getExtra(fieldEditor);
+            
+            // Add autocompleter listener, if required for this field:
+            /*
+            AutoCompleter<String> autoCompleter = bPanel.getAutoCompleters().get(field);
```

Comment:
Just a note: The auto completer is broken and therefore it is one of the few features that is fine to go.